docs/root/development/performance/binary_size.rst

```diff
@@ -0,0 +1,135 @@
+    1.3%  1.48Ki   1.3%  1.48Ki .data
+    0.0%       0   0.6%     672 .bss
+    0.1%      72   0.1%      72 .rodata
+
+Or one might want to see how sections of the binary map to compilation units::
+
+  $ bloaty --debug-file=bin/test_binary_size -c envoy.bloaty -d bloaty_package,compileunits,sections bin/test_binary_size.stripped
+  ...
+  13.2%   929Ki  13.0%   929Ki .rodata
+      81.2%   755Ki  81.2%   755Ki [section .rodata]
+      15.4%   143Ki  15.4%   143Ki boringssl/
+          37.9%  54.4Ki  37.9%  54.4Ki external/boringssl/src/crypto/obj/obj.c
+          21.7%  31.1Ki  21.7%  31.1Ki external/boringssl/src/third_party/fiat/curve25519.c
+          17.3%  24.9Ki  17.3%  24.9Ki external/boringssl/src/crypto/fipsmodule/bcm.c
+          11.3%  16.2Ki  11.3%  16.2Ki external/boringssl/err_data.c
+           4.5%  6.39Ki   4.5%  6.39Ki [55 Others]
+           0.9%  1.26Ki   0.9%  1.26Ki external/boringssl/src/crypto/x509v3/v3_crld.c
+           0.7%  1.06Ki   0.7%  1.06Ki external/boringssl/src/crypto/asn1/tasn_typ.c
+
+These different representations will give you perspective about how different changes in the binary will affect size.
+Note that the ``envoy.bloaty`` config refers to a bloaty config that has regexes to capture output.
+The example config used in this type of analysis is::
+
+  custom_data_source: {
+    name: "bloaty_package"
+    base_data_source: "compileunits"
+
+    #envoy source code.
+    rewrite: {
+      pattern: "^(external/envoy/source/)(\\w+/)(\\w+)"
+      replacement: "envoy \\2"
+    }
+
+    #envoy third party libraries.
+    rewrite: {
+        pattern: "^(external/)(\\w+/)"
+        replacement: "\\2"
+    }
+
+    #all compiled protos.
+    rewrite: {
+        pattern: "([.pb.cc | .pb.validate.cc])$"
+        replacement: "compiled protos"
+    }
+  }
+
+Open issues regarding size
+--------------------------
+
+
+``perf/size`` is a label tagging all current open issues that can improve binary size.
+Check out the issues `here <https://github.com/lyft/envoy-mobile/labels/perf%2Fsize>`_.
+After performing any change that tries to address these issues you should run through the analysis pipeline described above, and make sure your changes match expectations.
+
+The following issues are listed in priority order maximizing complexity vs. size win:
+
+- https://github.com/lyft/envoy-mobile/issues/174
+- https://github.com/lyft/envoy-mobile/issues/175
+- https://github.com/lyft/envoy-mobile/issues/180
+- https://github.com/lyft/envoy-mobile/issues/178
+- https://github.com/lyft/envoy-mobile/issues/177
+- https://github.com/lyft/envoy-mobile/issues/179
+- https://github.com/lyft/envoy-mobile/issues/176
+
+Current status
+~~~~~~~~~~~~~~
+
+As of https://github.com/lyft/envoy-mobile/tree/f17caebcfce09ec5dcda905dc8418fea4d382da7
+The test_binary_size_size as built by the toolchain against the architecture described above
+compiles to a stripped size of 8.9mb and a compressed size of 3mb.
```

Comment:
nice: this is actually a lot smaller than I thought, and I think we can make this smaller still.

Reply:
yep, lots of room